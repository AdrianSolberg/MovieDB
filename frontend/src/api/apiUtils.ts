import { Review } from "@/types/__generated__/types";

type ResponseReview = {
    __typename?: "Review";
    _id: string;
    username: string;
    rating: number;
    comment: string;
    date: Date;
};

/**
 *  Make sure that the date-field is correctly initialized as a Date-object
 *  when getting reviews as a response from a query.
 */
export function parseReviewResponse(inputReviews: ResponseReview[]): Review[] {
    return inputReviews.map((review) => ({
        ...review,
        date: new Date(review.date),
    })) as Review[];
}
