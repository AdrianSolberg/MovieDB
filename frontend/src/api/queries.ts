import { gql } from "@/types/__generated__";

export const GET_MOVIES = gql(`
    query GetMovies(
        $skip: Int
        $limit: Int
        $filters: FiltersInput
        $sortOption: SortingType
        $search: String
    ) {
        movies(
            skip: $skip
            limit: $limit
            filters: $filters
            sortOption: $sortOption
            search: $search
        ) {
            _id
            title
            vote_average
            release_date
            runtime
            poster_path
        }
    }
`);

export const GET_MOVIE = gql(`
    query GetMovie($movieId: Int!) {
        movie(id: $movieId) {
            _id
            title
            vote_average
            vote_count
            status
            release_date
            revenue
            runtime
            backdrop_path
            budget
            homepage
            imdb_id
            original_language
            original_title
            overview
            popularity
            poster_path
            tagline
            genres
            production_companies
            production_countries
            spoken_languages
            keywords
            reviews {
                _id
                username
                rating
                comment
                date
            }
        }
    }
`);

export const GET_FILTERS = gql(`
    query GetFilters {
        filters {
            Genre
            Rating
            Decade
            Status
            Runtime
        }
    }
`);

export const GET_LATEST_REVIEWS = gql(`
    query GetLatestReviews(
        $skip: Int,
        $limit: Int
    ) {
        latestReviews(
            skip: $skip,
            limit: $limit
        ) {
            _id
            movie {
                _id
                title
                poster_path
            }
            username
            rating
            comment
            date
        }
    }
`);

export const GET_USER_REVIEWS = gql(`
    query GetUserReviews(
        $username: String!,
        $skip: Int,
        $limit: Int
    ) {
        userReviews(
            username: $username,
            skip: $skip,
            limit: $limit
        ) {
            _id
            movie {
                _id
                title
                poster_path
            }
            rating
            comment
            date
        }
    }
`);

export const ADD_REVIEW = gql(`
    mutation AddReview(
        $movieId: Int!,
        $username: String!,
        $rating: Int!,
        $comment: String!
    ) {
        addReview(
            movie_id: $movieId,
            username: $username,
            rating: $rating,
            comment: $comment
        ) {
            _id
            vote_average
            vote_count
            reviews {
                _id
                username
                rating
                comment
                date
            }
        }
    }
`);

export const DELETE_REVIEW = gql(`
    mutation DeleteReview($id: ID!) {
        deleteReview(_id: $id) {
            _id
            vote_average
            vote_count
            reviews {
                _id
                username
                rating
                comment
                date
            }
        }
    }
`);
