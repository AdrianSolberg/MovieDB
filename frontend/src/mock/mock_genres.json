[
    {
        "id": 0,
        "name": "Drama"
    },
    {
        "id": 1,
        "name": "Romance"
    },
    {
        "id": 2,
        "name": "Western"
    },
    {
        "id": 3,
        "name": "Animation"
    },
    {
        "id": 4,
        "name": "War"
    },
    {
        "id": 5,
        "name": "Science Fiction"
    },
    {
        "id": 6,
        "name": "Comedy"
    },
    {
        "id": 7,
        "name": "Fantasy"
    },
    {
        "id": 8,
        "name": "Adventure"
    },
    {
        "id": 9,
        "name": "Crime"
    },
    {
        "id": 10,
        "name": "Thriller"
    },
    {
        "id": 11,
        "name": "Family"
    },
    {
        "id": 12,
        "name": "Action"
    },
    {
        "id": 13,
        "name": "Mystery"
    }
]
