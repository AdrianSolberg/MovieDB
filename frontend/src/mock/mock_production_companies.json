[
    "Lynda Obst Productions",
    "Pascal Pictures",
    "Fairview Entertainment",
    "Silver Pictures",
    "Walt Disney Pictures",
    "Visiona Romantica",
    "DC Films",
    "Mandate International",
    "Patalex IV Productions Limited",
    "Amblin Entertainment",
    "Heyday Films",
    "Universal Pictures",
    "Bron Studios",
    "Juno Pix",
    "Color Force",
    "LStar Capital",
    "Miramax",
    "The Saul Zaentz Company",
    "Sikelia Productions",
    "Red Granite Pictures",
    "The Weinstein Company",
    "Knickerbocker Films",
    "Columbia Pictures",
    "Groucho II Film Partnership",
    "Syncopy",
    "P of A Productions Limited",
    "Zehnte Babelsberg Film",
    "The Linson Company",
    "Pixar",
    "Appian Way",
    "The Steve Tisch Company",
    "Genre Films",
    "Paramount",
    "Kennedy Miller Productions",
    "Lucasfilm Ltd.",
    "New Line Cinema",
    "Atman Entertainment",
    "Cecchi Gori Pictures",
    "Castle Rock Entertainment",
    "Isobel Griffiths",
    "Phoenix Pictures",
    "DC Entertainment",
    "Dune Entertainment",
    "20th Century Fox",
    "Marvel Studios",
    "Lightstorm Entertainment",
    "WingNut Films",
    "Fox 2000 Pictures",
    "RatPac Entertainment",
    "Lionsgate",
    "Warner Bros. Pictures",
    "Patalex III Productions Limited",
    "Jersey Films",
    "Joint Effort",
    "Village Roadshow Pictures",
    "EMJAG Productions",
    "The Donners' Company",
    "1492 Pictures",
    "A Band Apart",
    "DC Comics",
    "Atlas Entertainment",
    "Ingenious Media",
    "Legendary Pictures",
    "Taurus Film",
    "Wendy Finerman Productions",
    "Peninsula Films",
    "Lin Pictures",
    "Regency Enterprises",
    "Jerry Bruckheimer Films"
]
