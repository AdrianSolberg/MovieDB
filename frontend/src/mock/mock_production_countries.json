[
    {
        "id": 0,
        "name": "New Zealand"
    },
    {
        "id": 1,
        "name": "Canada"
    },
    {
        "id": 2,
        "name": "South Africa"
    },
    {
        "id": 3,
        "name": "United Kingdom"
    },
    {
        "id": 4,
        "name": "Germany"
    },
    {
        "id": 5,
        "name": "Australia"
    },
    {
        "id": 6,
        "name": "France"
    },
    {
        "id": 7,
        "name": "United States of America"
    }
]
