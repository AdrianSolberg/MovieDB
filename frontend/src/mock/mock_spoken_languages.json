[
    {
        "id": 0,
        "name": "Swahili"
    },
    {
        "id": 1,
        "name": "French"
    },
    {
        "id": 2,
        "name": "German"
    },
    {
        "id": 3,
        "name": "Persian"
    },
    {
        "id": 4,
        "name": "Italian"
    },
    {
        "id": 5,
        "name": "Japanese"
    },
    {
        "id": 6,
        "name": "Korean"
    },
    {
        "id": 7,
        "name": "Hindi"
    },
    {
        "id": 8,
        "name": "Urdu"
    },
    {
        "id": 9,
        "name": "English"
    },
    {
        "id": 10,
        "name": "Mandarin"
    },
    {
        "id": 11,
        "name": "Spanish"
    },
    {
        "id": 12,
        "name": "Russian"
    },
    {
        "id": 13,
        "name": "Xhosa"
    },
    {
        "id": 14,
        "name": "Romanian"
    },
    {
        "id": 15,
        "name": "Norwegian"
    },
    {
        "id": 16,
        "name": "Swedish"
    },
    {
        "id": 17,
        "name": "Arabic"
    }
]
