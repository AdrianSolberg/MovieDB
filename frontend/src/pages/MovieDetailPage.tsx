import MovieCardDetailed from "../components/MovieCardDetailed";
import MovieReviews from "../components/MovieReviews";
import { Link, useParams } from "react-router-dom";
import { useQuery } from "@apollo/client";
import { Movie } from "@/types/__generated__/types";
import { GET_MOVIE } from "@/api/queries";
import { useMemo } from "react";
import Loader from "../components/Loader";

function MovieDetailPage() {
    const { movieId } = useParams<{ movieId: string }>();
    const movieIdAsInt = Number(movieId);

    const { data, loading, error } = useQuery(GET_MOVIE, {
        variables: { movieId: movieIdAsInt },
        skip: isNaN(movieIdAsInt),
    });

    const movie = useMemo(() => data?.movie as Movie | undefined, [data]);

    if (error) {
        return (
            <main className="mt-2 w-dvw text-center">
                <h1 className="text-2xl">Something went wrong!</h1>
                <Link to="/" className="text-primary hover:underline">
                    Return to home page
                </Link>
            </main>
        );
    }

    if (loading) {
        return (
            <main className="mt-2 w-dvw text-center">
                <Loader size="lg">
                    <p className="text-2xl">Loading...</p>
                </Loader>
            </main>
        );
    }

    if (!movie) {
        return (
            <main className="mt-2 w-dvw text-center">
                <h1 className="text-2xl">Could not find movie!</h1>
                <Link to="/" className="text-primary hover:underline">
                    Return to home page
                </Link>
            </main>
        );
    }

    return (
        <main id="main-content" className="pb-2">
            <MovieCardDetailed movie={movie} />
            <MovieReviews movie={movie} />
        </main>
    );
}

export default MovieDetailPage;
