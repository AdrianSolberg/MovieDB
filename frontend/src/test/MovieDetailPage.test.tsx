import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MockedProvider, MockedResponse } from "@apollo/client/testing";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import { GET_MOVIE } from "@/api/queries";
import MovieDetailPage from "@/pages/MovieDetailPage";
import { all_movies } from "./mock/util";

const movieMock = [
    {
        request: {
            query: GET_MOVIE,
            variables: {
                movieId: 475557,
            },
        },
        result: {
            data: {
                movie: all_movies[18],
            },
        },
    },
];

const errorMock = [
    {
        request: {
            query: GET_MOVIE,
            variables: { movieId: 475557 },
        },
        error: new Error("Error fetching movie"),
    },
];

const noMovieMock = [
    {
        request: {
            query: GET_MOVIE,
            variables: { movieId: 475557 },
        },
        result: { data: { movie: null } },
    },
];

describe("MovieDetailPage", () => {
    const renderComponent = (mocks: MockedResponse[] | undefined) => {
        const router = createMemoryRouter(
            [{ path: "/movie/:movieId", element: <MovieDetailPage /> }],
            {
                initialEntries: ["/movie/475557"],
            }
        );

        return render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <RouterProvider router={router} />
            </MockedProvider>
        );
    };

    it("matches snapshot when loading", () => {
        const { asFragment } = renderComponent(movieMock);
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot when data is loaded", async () => {
        const { asFragment } = renderComponent(movieMock);
        await screen.findByText("Joker");
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot when no movie is found", async () => {
        const { asFragment } = renderComponent(noMovieMock);
        await screen.findByText("Could not find movie!");
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot on error", async () => {
        const { asFragment } = renderComponent(errorMock);
        await screen.findByText("Something went wrong!");
        expect(asFragment()).toMatchSnapshot();
    });

    it("displays loading message initially", () => {
        renderComponent(movieMock);
        expect(screen.getByText("Loading...")).toBeInTheDocument();
    });

    it("displays movie details when data is successfully loaded", async () => {
        renderComponent(movieMock);
        expect(await screen.findByText("Joker")).toBeInTheDocument();
    });

    it("displays error message if there is a query error", async () => {
        renderComponent(errorMock);
        expect(
            await screen.findByText("Something went wrong!")
        ).toBeInTheDocument();
        expect(
            await screen.findByText("Return to home page")
        ).toBeInTheDocument();
    });

    it("displays 'Could not find movie!' message if movie is not found", async () => {
        renderComponent(noMovieMock);
        expect(
            await screen.findByText("Could not find movie!")
        ).toBeInTheDocument();
        expect(
            await screen.findByText("Return to home page")
        ).toBeInTheDocument();
    });
});
