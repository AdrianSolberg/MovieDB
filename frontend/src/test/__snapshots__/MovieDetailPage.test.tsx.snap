// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieDetailPage Snapshots > matches snapshot on error 1`] = `
<DocumentFragment>
  <section
    class="mt-2 w-dvw text-center"
  >
    <h1
      class="text-2xl"
    >
      Something went wrong!
    </h1>
    <a
      class="text-primary hover:underline"
      href="/"
    >
      Return to home page
    </a>
  </section>
</DocumentFragment>
`;

exports[`MovieDetailPage Snapshots > matches snapshot when data is loaded 1`] = `
<DocumentFragment>
  <section
    class="pb-2"
  >
    <div
      class="rounded-lg border bg-card text-card-foreground relative m-4 overflow-hidden bg-cover bg-center shadow-lg"
      data-testid="detailedMovieCard"
      style="background-image: url(https://image.tmdb.org/t/p/original/hO7KbdvGOtDdeg0W4Y5nKEHeDDh.jpg);"
    >
      <div
        class="absolute inset-0 bg-black opacity-85"
      />
      <div
        class="relative h-full p-6 text-white"
      >
        <div
          class="flex flex-col space-y-1.5 mb-4 p-0"
        >
          <h1
            class="tracking-tight text-5xl font-bold"
          >
            Joker
          </h1>
          <p
            class="text-sm"
          >
            <time
              datetime="2019"
            >
              2019
            </time>
             • 
            <time
              datetime="PT2H2M"
            >
              2h 2m
            </time>
             • 
            <span>
              Crime, Thriller, Drama
            </span>
          </p>
          <div
            class="flex items-center gap-2"
            title="4.08"
          >
            <svg
              class="lucide lucide-star fill-current text-yellow-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="-1"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <svg
              class="lucide lucide-star fill-current text-yellow-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="-1"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <svg
              class="lucide lucide-star fill-current text-yellow-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="-1"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <svg
              class="lucide lucide-star fill-current text-yellow-500"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="-1"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <div
              class="relative inline-block"
            >
              <svg
                class="lucide lucide-star fill-transparent text-yellow-500"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                />
              </svg>
              <div
                style="position: absolute; top: 0px; overflow: hidden; width: 8.399999999999963%;"
              >
                <svg
                  class="lucide lucide-star fill-current text-yellow-500"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <section
          class="flex h-full flex-col md:flex-row"
        >
          <figure
            class="mb-6 w-full max-w-72 flex-shrink-0 md:mb-0 md:mr-6"
          >
            <img
              alt="Poster of Joker"
              class="rounded-lg"
              src="https://image.tmdb.org/t/p/w500/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg"
              title="Joker"
            />
          </figure>
          <section
            class="flex-grow"
          >
            <p
              class="mb-2 text-xl italic text-gray-300"
            >
              Put on a happy face.
            </p>
            <p
              class="mb-4 text-lg"
            >
              During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.
            </p>
            <ul
              class="mb-6 flex flex-col gap-2"
            >
              <li>
                <strong>
                  Status: 
                </strong>
                Released
              </li>
              <li>
                <strong>
                  Budget: 
                </strong>
                $55,000,000
              </li>
              <li>
                <strong>
                  Revenue: 
                </strong>
                $1,074,458,282
              </li>
              <li>
                <strong>
                  Production Companies: 
                </strong>
                Warner Bros. Pictures, Joint Effort, Village Roadshow Pictures, Bron Studios, DC Films
              </li>
              <li>
                <strong>
                  Production Countries: 
                </strong>
                Canada, United States of America
              </li>
              <li>
                <strong>
                  Spoken Languages: 
                </strong>
                English
              </li>
            </ul>
            <a
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
              href="http://www.jokermovie.net/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Visit Homepage
            </a>
          </section>
        </section>
      </div>
    </div>
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm m-4"
    >
      <section
        class="m-4 mb-6"
      >
        <h2
          class="mb-4 text-2xl font-bold"
        >
          Submit review
        </h2>
        <form
          class="space-y-4"
        >
          <div
            class="flex items-center gap-2"
            role="radiogroup"
          >
            <svg
              aria-checked="false"
              aria-label="1 star"
              class="lucide lucide-star text-yellow-200 cursor-pointer"
              fill="none"
              height="24"
              role="radio"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <svg
              aria-checked="false"
              aria-label="2 star"
              class="lucide lucide-star text-yellow-200 cursor-pointer"
              fill="none"
              height="24"
              role="radio"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <svg
              aria-checked="false"
              aria-label="3 star"
              class="lucide lucide-star text-yellow-200 cursor-pointer"
              fill="none"
              height="24"
              role="radio"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <svg
              aria-checked="false"
              aria-label="4 star"
              class="lucide lucide-star text-yellow-200 cursor-pointer"
              fill="none"
              height="24"
              role="radio"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
            <svg
              aria-checked="false"
              aria-label="5 star"
              class="lucide lucide-star text-yellow-200 cursor-pointer"
              fill="none"
              height="24"
              role="radio"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              tabindex="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
              />
            </svg>
          </div>
          <textarea
            class="flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
            id="review-comment"
            maxlength="1500"
            name="comment"
            placeholder="Write your review (optional)"
          />
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
            disabled=""
            type="submit"
          >
            Submit Review
          </button>
        </form>
      </section>
      <section
        class="m-4 space-y-4"
      >
        <ul
          class="space-y-6"
        />
      </section>
    </div>
  </section>
</DocumentFragment>
`;

exports[`MovieDetailPage Snapshots > matches snapshot when loading 1`] = `
<DocumentFragment>
  <section
    class="mt-2 w-dvw text-center"
  >
    <div
      aria-live="polite"
      class="m-2 flex flex-col items-center justify-center gap-2"
      role="status"
    >
      <svg
        class="animate-spin h-16 w-16 stroke-primary"
        viewBox="0 0 800 800"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="400"
          cy="400"
          fill="none"
          r="350"
          stroke-dasharray="1648,550"
          stroke-linecap="round"
          stroke-width="64"
        />
      </svg>
      <p
        class="text-2xl"
      >
        Loading...
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`MovieDetailPage Snapshots > matches snapshot when no movie is found 1`] = `
<DocumentFragment>
  <section
    class="mt-2 w-dvw text-center"
  >
    <h1
      class="text-2xl"
    >
      Could not find movie!
    </h1>
    <a
      class="text-primary hover:underline"
      href="/"
    >
      Return to home page
    </a>
  </section>
</DocumentFragment>
`;
